receiver/mongodbatlasreceiver/mongodb_event_to_logdata.go

```diff
@@ -0,0 +1,167 @@
+	"encoding/json"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbatlasreceiver/internal/model"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+)
+
+const (
+	// Number of log attributes to add to the plog.LogRecordSlice.
+	totalLogAttributes = 5
+
+	// Number of resource attributes to add to the plog.ResourceLogs.
+	totalResourceAttributes = 4
+)
+
+const layout = "2006-01-02T15:04:05.000-07:00"
+
+// Only two types of events are created as of now.
+// For more info: https://docs.openshift.com/container-platform/4.9/rest_api/metadata_apis/event-core-v1.html
+var severityMap = map[string]plog.SeverityNumber{
+	"F":  plog.SeverityNumberFATAL,
+	"E":  plog.SeverityNumberERROR,
+	"W":  plog.SeverityNumberWARN,
+	"I":  plog.SeverityNumberINFO,
+	"D":  plog.SeverityNumberDEBUG,
+	"D1": plog.SeverityNumberDEBUG,
+	"D2": plog.SeverityNumberDEBUG2,
+	"D3": plog.SeverityNumberDEBUG3,
+	"D4": plog.SeverityNumberDEBUG4,
+	"D5": plog.SeverityNumberDEBUG4,
+}
+
+// mongoAuditEventToLogRecord converts model.AuditLog event to plog.LogRecordSlice and adds the resource attributes.
+func mongodbAuditEventToLogData(logger *zap.Logger, e *model.AuditLog, r resourceInfo) plog.Logs {
+	ld := plog.NewLogs()
+	rl := ld.ResourceLogs().AppendEmpty()
+	sl := rl.ScopeLogs().AppendEmpty()
+	lr := sl.LogRecords().AppendEmpty()
+
+	resourceAttrs := rl.Resource().Attributes()
+	resourceAttrs.EnsureCapacity(totalResourceAttributes)
+
+	// Attributes related to the object causing the event.
+	resourceAttrs.InsertString("org", r.Org.Name)
+	resourceAttrs.InsertString("project", r.Project.Name)
+	resourceAttrs.InsertString("cluster", r.Cluster.Name)
+	resourceAttrs.InsertString("hostname", r.Hostname)
+
+	data, err := json.Marshal(e)
+	if err != nil {
+		logger.Warn("failed to marshal", zap.Error(err))
+	}
+
+	t, err := time.Parse(layout, e.Timestamp.Date)
+	if err != nil {
+		logger.Warn("Time failed to parse correctly", zap.Error(err))
+	}
+	lr.SetTimestamp(pcommon.NewTimestampFromTime(t))
+	lr.SetObservedTimestamp(pcommon.NewTimestampFromTime(time.Now()))
+
+	// The Message field contains description about the event,
+	// which is best suited for the "Body" of the LogRecordSlice.
+	lr.Body().SetStringVal(string(data))
+
+	// Set the "SeverityNumber" and "SeverityText" if a known type of
+	// severity is found.
+	lr.SetSeverityNumber(plog.SeverityNumberINFO)
+	lr.SetSeverityText("INFO")
+
+	attrs := lr.Attributes()
+	attrs.EnsureCapacity(totalLogAttributes)
+
+	if e.AuthType != "" {
+		attrs.InsertString("authtype", e.AuthType)
+	}
+
+	attrs.InsertString("local.ip", e.Local.IP)
+	attrs.InsertInt("local.port", int64(e.Local.Port))
+	attrs.InsertString("remote.ip", e.Remote.IP)
+	attrs.InsertInt("remote.port", int64(e.Remote.Port))
+	attrs.InsertString("uuid.binary", e.ID.Binary)
+	attrs.InsertString("uuid.type", e.ID.Type)
+	attrs.InsertInt("remote.port", int64(e.Result))
```

Comment:
Looks like this is inserted twice, I think you meant to insert a different key here?